mlflow/azure/client.py

```diff
@@ -14,6 +14,84 @@
 }
 
 
+def put_adls_file_creation(sas_url, headers):
+    """
+    Performs an ADLS Azure file create `Put` operation
+    (https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/create)
+
+    :param sas_url: A shared access signature URL referring to the Azure ADLS server
+                    to which the file creation command should be issued.
+    :param headers: Additional headers to include in the Put request body
+    """
+    request_url = _append_query_parameters(sas_url, {"resource": "file"})
+
+    request_headers = {}
+    for name, value in headers.items():
+        if _is_valid_adls_put_header(name):
+            request_headers[name] = value
+        else:
+            _logger.debug("Removed unsupported '%s' header for ADLS Gen2 Put operation", name)
+
+    with rest_utils.cloud_storage_http_request(
+        "put", request_url, headers=request_headers
+    ) as response:
+        rest_utils.augmented_raise_for_status(response)
+
+
+def patch_adls_file_upload(sas_url, data, position, headers, is_single):
+    """
+    Performs an ADLS Azure file create `Patch` operation
+    (https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/update)
+
+    :param sas_url: A shared access signature URL referring to the Azure ADLS server
+                    to which the file update command should be issued.
+    :param data: Data to include in the Patch request body.
+    :param position: Positional offset of the data in the Patch request
+    :param headers: Additional headers to include in the Patch request body
+    """
+    new_params = {"action": "append", "position": str(position)}
+    if is_single:
+        new_params["flush"] = "true"
```

Comment:
Why do we set `flush` to `true` only when we have a single chunk? What happens if we have multiple chunks and set `flush` to `true` in the final append?


```
append
append
append with flush
```

Reply:
I believe this is a shortcut that microsoft added to reduce the number of calls in a single part upload.  My understanding is this only works when the position is 0 for a single part upload.